cpp/models/abm/testing_strategy.cpp

```diff
@@ -105,12 +102,20 @@ void TestingScheme::update_activity_status(TimePoint t)
 
 bool TestingScheme::run_scheme(PersonalRandomNumberGenerator& rng, Person& person, TimePoint t) const
 {
-    if (t - person.get_time_of_last_test() > m_minimal_time_since_last_test) {
-        if (m_testing_criteria.evaluate(person, t)) {
-            double random = UniformDistribution<double>::get_instance()(rng);
-            if (random < m_probability) {
-                return !person.get_tested(rng, t, m_test_parameters);
-            }
+    auto test_result = person.get_test_result(m_test_parameters.type);
+    // If the agent has a test result valid until now, use the result directly
+    if ((test_result.type != TestType::Count) &&
```

Comment:
Do we need to check for this? Perhaps an `assert` would be a solution?

Reply:
Yes, I updated it. The main idea is now to see if the `test_result.time_of_testing != TimePoint(std::numeric_limits<int>::min())`, i.e. a new test is added.